spec/cohort.spec.js

```diff
@@ -0,0 +1,15 @@
+import Cohort from '../src/cohort.js'
+
+describe('cohort', () => {
+  describe('creation', () => {
+    it('is possible with valid name input', () => {
+      const myCohort = new Cohort('#11')
+      expect(myCohort.name).toEqual('#11')
+      expect(myCohort.students).toEqual([])
+    })
+
+    it('creation fails when name is not a string', () => {
+      expect(() => new Cohort('')).toThrowError('invalid input')
```

Comment:
another way I was testing to write these expect statements was to extract out the functionality into a variable like so:

const expectation = 'invalid input'
const action = () => new Cohort('')

expect(action).toThrowError(expectation)

Just for fun.

Reply:
I saw that in Ed's repo, and have to admit i am still a bit shaky in understanding why you don't CALL action to get anything in return, is it just omission of input variables as there are none? it seems wildly inconsistent of javascript to distinguish between functions and function calls and then turn around by allowing to imply "oh yea expect(action) will call action lol"

Reply:
I agree. I can get mega lost in these as well.